janus_core/calculations/eos.py

```diff
@@ -0,0 +1,117 @@
+        Maximum lattice constant scale factor. Default is 1.05.
+    n_lattice : int
+        Number of lattice constants to use. Default is 5.
+    minimize : bool
+        Whether to optimize geometry. Default is False.
+    minimize_kwargs : Optional[dict[str, Any]]
+        Keyword arguments to pass to optimize. Default is None.
+    file_prefix : Optional[PathLike]
+        Prefix for output filenames. Default is inferred from structure name, or
+        chemical formula of the structure.
+    log_kwargs : Optional[dict[str, Any]]
+        Keyword arguments to pass to `config_logger`. Default is {}.
+
+    Returns
+    -------
+    EquationOfState
+        Equation of state ASE object for structure.
+    """
+    [log_kwargs] = none_to_dict([log_kwargs])
+    log_kwargs.setdefault("name", __name__)
+    logger = config_logger(**log_kwargs)
+
+    struct_name = struct_name if struct_name else struct.get_chemical_formula()
+    file_prefix = file_prefix if file_prefix else struct_name
+
+    traj_file = f"{file_prefix}-eos.traj"
+
+    if not struct.calc:
+        raise ValueError("Please attach a calculator to `struct`.")
+
+    # Ensure lattice constants span correct range
+    if n_lattice <= 1:
+        raise ValueError("`n_lattice` must be greater than 1.")
+    if not 0 < min_lattice < 1:
+        raise ValueError("`min_lattice` must be between 0 and 1.")
+    if max_lattice <= 1:
+        raise ValueError("`max_lattice` must be greater than 1.")
+
+    if minimize:
+        if logger:
+            minimize_kwargs["log_kwargs"] = {
+                "filename": log_kwargs["filename"],
+                "name": logger.name,
+                "filemode": "a",
+            }
+        optimize(struct, **minimize_kwargs)
+
+    cell = struct.get_cell()
+
+    traj = Trajectory(traj_file, "w")
+    lattice_scalars = np.linspace(min_lattice, max_lattice, n_lattice)
+    for lattice_scalar in lattice_scalars:
+        struct.set_cell(cell * lattice_scalar, scale_atoms=True)
+        struct.get_potential_energy()
+        traj.write(struct)
+
+    # Read n_structs configurations
+    configs = read(f"{traj_file}@0:{n_lattice}")
+
+    # Extract volumes and energies:
+    volumes = [config.get_volume() for config in configs]
+    energies = [config.get_potential_energy() for config in configs]
+
+    with open(f"{file_prefix}-eos-raw.dat", "w", encoding="utf8") as out:
+        print("#Lattice Scalar | Energy [eV] | Volume [Å^3] ", file=out)
+        for lattice_scalar, energy, volume in zip(lattice_scalars, energies, volumes):
+            print(f"{lattice_scalar} {energy} {volume}", file=out)
+
+    eos = EquationOfState(volumes, energies)
+
+    v_0, e_0, bulk_modulus = eos.fit()
+    bulk_modulus *= 1.0e24 / kJ
+    with open(f"{file_prefix}-eos-fit.dat", "w", encoding="utf8") as out:
+        print("#B [GPa] | Energy [eV] | Volume [Å^3] ", file=out)
+        print(f"{bulk_modulus} {e_0} {v_0}", file=out)
+
+    return eos
```

Comment:
```suggestion
    return eos, bulk_modulus, v_0, e_0
```

Reply:
Might it be nicer to return this as a results dictionary, rather than a tuple?